### PR TITLE
Fix attaching volume container

### LINKED ISSUE #759: failed to create container with attached volume in a single reservation
https://explorer.testnet.grid.tf/explorer/reservations/459798

### PR DESCRIPTION
**Pull request by DylanVerstraete**

### CODE REVIEW

File: pkg/provision/primitives/converter.go [hunk 1]
```diff
@@ -42,6 +43,9 @@ func ContainerToProvisionType(c workloads.Container) (Container, string, error)
 	}
 
 	for i, mount := range c.Volumes {
+		if strings.HasPrefix(mount.VolumeId, "-") {
+			mount.VolumeId = reservationID + mount.VolumeId
```
Comment (zaibon) [inline pkg/provision/primitives/converter.go:47]: The format needs to be `reservationID-workloadsID`
So 
```suggestion
			mount.VolumeId = fmt.Sprintf("%s-%s", reservationID, mount.VolumeId)
```

File: pkg/provision/primitives/converter_test.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func TestTfgridReservationContainer1_ToProvisionType(t *testing.T) {
 				NetworkConnection: tt.fields.NetworkConnection,
 				StatsAggregator:   tt.fields.StatsAggregator,
 			}
-			got, _, err := ContainerToProvisionType(c)
+			got, _, err := ContainerToProvisionType(c, "")
```
Comment (zaibon) [inline pkg/provision/primitives/converter_test.go]: can we add a test case that test the new code too  ?

**Review by zaibon — CHANGES_REQUESTED** [review]

File: pkg/provision/primitives/converter.go (hunk 1 above)
Comment (DylanVerstraete) [reply pkg/provision/primitives/converter.go:47]: this is not needed since it passes the check if it has a prefix. So just concatinating will work?

Comment (zaibon) [reply pkg/provision/primitives/converter.go:47]: right, my bad :-P

**Review by zaibon — APPROVED** [review]

**Pull request merged**